PedroChaparro/loomies-backend
# Feat: Fuse loomies

### Includes: 

- [x] Check the user owns the two Loomies.
- [x] Check both Loomies are "the same", Eg. Two "Caterpillars". 
- [x] Keep the max stats between the two Loomies.
- [x] Add the Level field on the caught_loomies collection.
- [x] Increment the level of the "resulting Loomie". 
- [x] Update the Swagger docs.

### How can I test it? 

After:

1. Re-generate the database (Except the users table). 
2. Copy some Ids from the `caught_loomies` collection to the `loomies` array of your desired user. **Make sure you copy, at least, two loomies with the same `serial` field to be able to fuse them.**
3. (Optional) Increment the `experience` field of the Loomies you'll fuse. You can find the required experience f/e level on notion (Pokémon Data page). 
4. Make a `POST` request to the new endpoint following the Swagger docs. 

Before: 

5. The Loomie with the lesser Level should be removed from the `caught_loomies` collection and the user's inventory. 
6. If you increment the `experience` of the loomies, you should see that the "new loomie" `experience` is now the sum of the individual experiences and possibly, the `level` field of the loomie had been also incremented.

---

**IMPORTANT ❗❗**

Before testing this PR, update the `loomie_team` field of your user to an empty array (`[ ]`) instead of the current `null` value. That was fixed on #83.
